packages/cyberstorm/src/components/LinkingProvider.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Implementation-agnostic context provider for navigation.
+ *
+ * @thunderstore/components aims to be reusable. Therefore the links
+ * used in the components, by default, render nothing. It's up to the
+ * app using the components to define how the links are rendered and
+ * inject the definitions with a context provider that wraps the
+ * components:
+ *
+ * 1. Create "myLinkLib" that implements the provided LinkLibrary interface.
+ * 2. Wrap the components with <LinkingProvider value={myLinkLib}>.
+ * 3. Components automatically use myLinkLib to determine how they
+ *    should render any links the contain.
+ *
+ * To add a new link type, follow the 4 required steps defined below.
+ */
+import React, { ReactElement as RE } from "react";
+
+// STEP 1 of adding new link definitions:
+// If the new link uses any props not already listed here, add them.
+export interface ThunderstoreLinkProps {
+  community?: string;
+  namespace?: string;
+  package?: string;
+  team?: string;
+}
+
+// STEP 2 of adding new link definitions:
+// If any new properties were added to ThunderstoreLinkProps, add a
+// dummy value for them here (the value doesn't matter). This is used to
+// strip the props so they don't get passed to HTML elements in React
+// implementations. Second, redundant declaration as an object is
+// required since the stripping is done during runtime.
+export const thunderstoreLinkProps: ThunderstoreLinkProps = {
+  community: "",
+  namespace: "",
+  package: "",
+  team: "",
+};
+
+// Accepting any and all props is required to keep the linking
+// implementation-agnostic. E.g. an implementation with Chakra has all
```

Comment:
Not sure we should refer to Chakra here anymore.

Reply:
oopsie, should be fixed now